apis/quay/v1/quayregistry_types.go

```diff
@@ -88,6 +101,18 @@ type Component struct {
 	// Managed indicates whether or not the Operator is responsible for the lifecycle of this component.
 	// Default is true.
 	Managed bool `json:"managed"`
+	// Overrides holds information regarding component specific configurations.
+	Overrides Override `json:"overrides,omitempty"`
+}
+
+// Override describes configuration overrides for the given managed component
+type Override struct {
+	// Env holds environment variable overrides that will be added to the given managed components deployment (if available)
+	Env map[string]string `json:"env,omitempty"`
```

Comment:
I think we might consider using [EnvVar](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L1981-L2001) type here. WDYT?

Reply:
Removed env implementation for this PR